keepers/validator-keeper/src/mev_commission.rs

```diff
@@ -215,3 +285,38 @@ async fn get_existing_entries(
     // Fetch existing tip distribution accounts for this epoch
     Ok(result)
 }
+
+async fn get_entries_with_uploaded_merkleroot(
+    client: Arc<RpcClient>,
+    entries: &[ValidatorMevCommissionEntry],
+) -> Result<Vec<ValidatorMevCommissionEntry>, MultipleAccountsError> {
+    /* Filters tip distribution tuples to the addresses, then fetches accounts to see which ones have an uploaded merkle root */
+    let tip_distribution_addresses = entries
+        .iter()
+        .map(|entry| entry.tip_distribution_account)
+        .collect::<Vec<Pubkey>>();
+
+    let accounts = get_multiple_accounts_batched(&tip_distribution_addresses, &client).await?;
+    let result = accounts
+        .iter()
+        .enumerate()
+        .filter_map(|(i, account_data)| {
+            if let Some(account_data) = account_data {
+                let mut data: &[u8] = &account_data.data;
+                if let Ok(tda) = TipDistributionAccount::try_deserialize(&mut data) {
+                    if let Some(_) = tda.merkle_root {
+                        return Some(entries[i].clone());
+                    } else {
+                        return None;
+                    }
+                } else {
+                    return None;
+                }
+            } else {
+                return None;
+            }
```

Comment:
nit: could get rid of all the `else`s and just set the last line as `None`

Reply:
yeah good call, done